### ISSUE TITLE
Add `ravel` / `flatten`

### ISSUE THREAD

**Issue by dlwh** [stanford-crfm/haliax#18]
Pretty sure the syntax should be

```python
a.ravel(new_axis_name: AxisSelector)
```

**Comment by rohan-mehta-1024** [comment]
I was taking a look at this and wrote up a simple implementation that just calls flatten_axes. However I saw this comment in the code which seems to have a different type signature

```
TODO: implement ravel. Can only do if we either ask for an axis or add ProductAxis or something
 def ravel(self, order='C') -> Any:
```

Which should the method follow? And should there be support for the optional order argument?

**Comment by dlwh** [comment]
don't worry about that old todo. I had a very different vision early on in Haliax and that was what I was thinking at the time.

Thanks!

**Comment by dlwh** [comment]
Fixed in #33 . Thanks @rohan-mehta-1024 !
